src/reducers/gifs.js

```diff
@@ -0,0 +1,100 @@
+import gifsActions from '../actions/gifs';
+
+const initialState = {
+  favorites: {},
+  gifs: {},
+  favoritesResults: {},
+  searchResults: {},
+  loading: false,
+  searchQuery: false
+};
+
+const { types } = gifsActions;
+
+const gifsReducer = (state = initialState, action) => {
+  const { type, payload } = action;
+
+  switch (type) {
+    case types.LOAD_DATA_START: {
+      return { ...state, loading: true };
+    }
+    case types.LOAD_DATA_FINISHED: {
+      const {
+        gifs: {
+          data: { data: gifs }
+        }
+      } = payload;
+
+      const mappedGifs = gifs.reduce((acc, gif) => {
+        acc[gif.id] = gif;
+        return acc;
+      }, {});
+
+      return { ...state, gifs: mappedGifs, loading: true };
+    }
+    case types.LOAD_DATA_FAILED: {
```

Comment:
I think this reducer is too big, it handles different business logic. Always try to make them as atomic as possible 👍